### ISSUE TITLE
How do I add another run time library to the toolchain?

### ISSUE THREAD

**Issue by AaronPinto** [wpilibsuite/roborio-toolchain#14]
So I'm trying to build the Android Debug Bridge from source for the RoboRIO and I've been using [this repo](https://github.com/bonnyfone/adb-arm/) with the Linux 2020.2 Rio toolchain. I'm doing all of this on Ubuntu WSL and it gets through most of the adb-download-make.sh script until the Make fails with the error "cannot find -lcrypto". 
![image](https://user-images.githubusercontent.com/10177767/73138967-8d39ab80-4036-11ea-9aa4-92a94141e312.png)
I'm assuming this is because it's not included in the released Rio toolchain so I have to rebuild the toolchain. 

I'm a bit new to this toolchain stuff so I was wondering if anyone could tell me what exactly I need to do to add another library to the toolchain and then build it myself. I'm also assuming that I need to add the libcrypto1.0.2 library [found here](http://download.ni.com/ni-linux-rt/feeds/2019/arm/cortexa9-vfpv3/) to the download.sh file, but I don't know what to do after that.

Any help is greatly appreciated, thanks!

**Comment by ThadHouse** [comment]
There isn't really a great way to add more libraries. It was worked on being added to the raspberry pi compiler, but never got moved over here.

However, you can likely add that library without recompiling the compiler. If you download the library (you have the right link that's where everything is located) you should be able to extract it and manually put the libraries in the usr/lib folder of the compiler. That would do the same thing as recompiling the compiler without actually needing to do so.

**Comment by AaronPinto** [comment]
So that would be in `/mnt/d/frc2020/roborio/arm-frc2020-linux-gnueabi/usr/lib` ?

**Comment by PeterJohnson** [comment]
Or you can download them from the NI repo.  See download.sh for some of the URLs (it’s browseable).

**Comment by AaronPinto** [comment]
Ok so I got the .ipk from the NI repo, and now do I just extract all of it in the `frc2020/roborio/arm-frc2020-linux-gnueabi/usr/lib` path? And I'm assuming I don't need the debian binary that comes with the .ipk.

**Comment by AaronPinto** [comment]
Ok, I believe I figured it out. So I extracted the .ipk to `frc2020/roborio/arm-frc2020-linux-gnueabi/usr/lib`, deleted `control.tar.gz` and `debian-binary`, and extracted `data.tar.gz`. When the extraction finished, `libcrypto.so.1.0.2` was located in `frc2020/roborio/arm-frc2020-linux-gnueabi/usr/lib/usr/lib` so I moved that to `frc2020/roborio/arm-frc2020-linux-gnueabi/usr/lib`. 

Then I ran `adb-download-make.sh` and it gave me the same error, so I googled it and found [this StackOverflow answer](https://stackoverflow.com/a/335968/6713362), so I modified `makefile.sample` line 69 from `-lcrypto` to `-l:libcrypto.so.1.0.2`. I re-ran `adb-download-make.sh` and it worked.
